tests/trading/test_wcl_fuzzy.py

```diff
@@ -89,33 +94,35 @@ def acquireShortShareSet(outcome, amount, approvalAddress, sender):
     assert order[SHARES_ESCROWED] == creatorLongShares or creatorShortShares
 
     # fill order
+    fillerEthRequiredLong = 0 if fillerLongShares == 0 else fillerLongShares * market.getNumTicks()
+    fillerEthRequiredShort = 0 if fillerShortShares == 0 else fillerShortShares * market.getNumTicks()
     acquireLongShares(orderOutcome, fillerLongShares, fillOrder.address, sender = fillerKey)
     acquireShortShareSet(orderOutcome, fillerShortShares, fillOrder.address, sender = fillerKey)
     remaining = fillOrder.publicFillOrder(orderID, orderSize, 42, sender = fillerKey, value = fillerTokens)
     assert not remaining
 
     # assert final state
-    assert cash.balanceOf(creatorAddress) == expectedMakerTokens
-    assert cash.balanceOf(fillerAddress) == expectedFillerTokens
+    assert fixture.chain.head_state.get_balance(creatorAddress) == creatorOriginalBalance - creatorEthRequiredLong - creatorEthRequiredShort - creatorTokens + expectedMakerTokens
+    assert fixture.chain.head_state.get_balance(fillerAddress) == fillerOriginalBalance - fillerEthRequiredLong - fillerEthRequiredShort - fillerTokens + expectedFillerTokens
     for outcome in range(0, market.getNumberOfOutcomes()):
-        shareToken = contractsFixture.applySignature('ShareToken', market.getShareToken(outcome))
+        shareToken = fixture.applySignature('ShareToken', market.getShareToken(outcome))
         if outcome == orderOutcome:
             assert shareToken.balanceOf(creatorAddress) == expectedMakerLongShares
             assert shareToken.balanceOf(fillerAddress) == expectedFillerLongShares
         else:
             assert shareToken.balanceOf(creatorAddress) == expectedMakerShortShares
             assert shareToken.balanceOf(fillerAddress) == expectedFillerShortShares
 
-def execute_bidOrder_tests(contractsFixture, universe, market, fxpAmount, fxpPrice):
-    longCost = long(fxpAmount * fxpPrice / 10**18)
-    shortCost = long(fxpAmount * (market.getNumTicks() - fxpPrice) / 10**18)
-    completeSetFees = long(fxpAmount * market.getNumTicks() * fix('0.0101') / 10**18 / 10**18)
+def execute_bidOrder_tests(fixture, kitchenSinkSnapshot, universe, market, fxpAmount, fxpPrice):
+    longCost = long(fxpAmount * fxpPrice)
+    shortCost = long(fxpAmount * (market.getNumTicks() - fxpPrice))
+    completeSetFees = long(fxpAmount * market.getNumTicks() * fix('0.0101') / 10**18)
     shortFee = long((completeSetFees * shortCost) / (longCost + shortCost))
     longFee = completeSetFees - shortFee
 
-    print "creator escrows cash, filler pays with cash"
+    print "creator escrows ETH, filler pays with ETH"
```

Comment:
There should be a reset to snapshot call before this first execute, otherwise it will run with whatever state the last test to run was in.

Consider moving the `fixture.resetToSnapshot(kitchenSinkSnapsho)` into the `execute` method to ensure that it is run before every iteration, without having to add it before each of the calls to execute.  Note: This will require adding the `kitchenSinkSnapshot` as a parameter to `execute`.

Reply:
Sounds good.  I'll move the reset line into `execute`.